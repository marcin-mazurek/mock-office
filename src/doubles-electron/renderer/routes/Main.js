import React from 'react';
import SideBar from '../sidebar/SideBar';

const Main = ({ children }) => (
  <div className="main">
    <div className="main__navbar">
      <SideBar />
    </div>
    <div className="main__content">
      {children}
    </div>
  </div>
);

Main.propTypes = {
  children: React.PropTypes.node
};

export default Main;
