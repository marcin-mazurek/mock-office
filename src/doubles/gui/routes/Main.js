import React from 'react';
import PropTypes from 'prop-types';
import SideBar from '../sidebar/SideBar';

const Main = ({ children }) => (
  <div className="main">
    <div className="main__navbar">
      <SideBar />
    </div>
    <div className="main__content">
      {children}
    </div>
  </div>
);

Main.propTypes = {
  children: PropTypes.node
};

export default Main;
