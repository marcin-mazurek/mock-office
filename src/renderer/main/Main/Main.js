import React from 'react';
import { ipcRenderer } from 'electron';
import UnloadedMockList from '../MockList/UnloadedMockList';
import LoadedMockList from '../MockList/LoadedMockList';
import FilePicker from '../../mocks/addFromFile/FilePicker';
import { SERVER_START, SERVER_STOP } from '../../../common/messageNames';

export default class Main extends React.Component {
  constructor(props) {
    super(props);
    this.toggleLive = this.toggleLive.bind(this);
    this.state = {
      live: false
    };
  }

  toggleLive() {
    const currentLive = !this.state.live;
    const message = currentLive ? SERVER_START : SERVER_STOP;
    ipcRenderer.send(message);

    this.setState({
      live: currentLive
    });
  }

  render() {
    return (
      <div className="main">
        <FilePicker />
        <div className="main-columns">
          <div>
            <h2>Mocks:</h2>
            <UnloadedMockList buttonText="Load" />
          </div>
          <div>
            <h2>Loaded:</h2>
            <LoadedMockList buttonText="Unload" />
          </div>
        </div>
        <button className="main" onClick={this.toggleLive}>
          {
            this.state.live
              ? 'Stop'
              : 'Start'
          }
        </button>
      </div>
    );
  }
}

Main.propTypes = {};
