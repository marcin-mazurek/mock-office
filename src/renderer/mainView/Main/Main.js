import React from 'react';
import { connect } from 'react-redux';
import Server from '../Server';
import ServerList from '../ServerList';
import { getSelected } from '../../servers/selectors';
import AddServer from '../../addServer/AddServer';

class Main extends React.Component {
  render() {
    const { selected } = this.props;

    return (
      <div className="main">
        <div className="main-server-list">
          <div>List of servers:</div>
          <ServerList />
        </div>
        <div>
          <AddServer />
          {
            selected
              ? (
                <Server
                  toggleLive={this.toggleLive}
                />
              )
              : null
          }
        </div>
      </div>
    );
  }
}

Main.propTypes = {
  selected: React.PropTypes.string
};

const mapStateToProps = state => ({
  selected: getSelected(state)
});

export default connect(mapStateToProps)(Main);
