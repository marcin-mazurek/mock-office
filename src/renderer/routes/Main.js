import React from 'react';
import NavBar from '../app/NavBar';

const Main = ({ children }) => (
  <div className="main">
    <div className="main__nav-bar">
      <NavBar />
    </div>
    <div className="main__content">
      {children}
    </div>
  </div>
);

Main.propTypes = {
  children: React.PropTypes.node
};

export default Main;
