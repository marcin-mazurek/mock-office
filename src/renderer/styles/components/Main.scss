.main {
  height: 100%;
  width: 100%;
}

.main-server-list {
  border-right: 1px solid #ddd;
}
