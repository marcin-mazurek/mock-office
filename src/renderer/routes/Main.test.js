import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Main from './Main';

test('Main should render', () => {
  const wrapper = shallow(<Main />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
