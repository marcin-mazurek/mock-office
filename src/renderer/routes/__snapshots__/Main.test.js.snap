exports[`test Main should render 1`] = `
<div
  className="main">
  <div
    className="main__navbar">
    <SideBar />
  </div>
  <div
    className="main__content" />
</div>
`;
